internal/provider/common.go

```diff
@@ -115,15 +113,27 @@ func ValidateImage(image string, metadata, labels map[string]string, watchByDef
 
 	// Update provider metadata with metadata from img labels
 	if err := mergo.Merge(&img.Metadata, metadata); err != nil {
-		return img, fmt.Errorf("failed merging metadata: %w", err)
+		return img, errors.Wrapf(err, "failed merging metadata")
 	}
 
 	return img, nil
 }
 
 func validateMetadataKey(key string) error {
 	if !metadataKeyRegexp.MatchString(key) {
-		return fmt.Errorf("only %q are allowed", metadataKeyChars)
+		return errors.Errorf("only %q are allowed", metadataKeyChars)
 	}
 	return nil
 }
+
+type invalidLabelError struct {
+	error
+}
+
+func (e *invalidLabelError) Error() string {
+	return e.Error()
+}
+
+func (e *invalidLabelError) Unwrap() error {
+	return e
+}
```

Comment:
@IamTheFij Fyi

Reply:
Why is this needed? If you wrap an error it should automatically have an `Unwrap()` method and support `Is()`

In the status quo, using the `%w` directive performed this wrapping and allowed the use of `errors.Unwrap(err)` already. Eg. https://go.dev/play/p/NqjuX4qfVmd